package_test.go

```diff
@@ -22,25 +22,40 @@ type PackageSuite struct{}
 
 var _ = Suite(&PackageSuite{})
 
-func setupDB() (*sql.DB, error) {
-	return sql.Open("sqlite3", "file:test.db?cache=shared&mode=memory")
-}
+func (s *PackageSuite) TearDownTest(c *C) {
+	// Try to run finalizers by calling GC several times.
+	for i := 0; i <= 10; i++ {
+		runtime.GC()
+		time.Sleep(0)
+	}
 
-func createExampleDB(createTables string, inserts []string) (*sql.DB, error) {
-	db, err := setupDB()
-	if err != nil {
-		return nil, err
+	stmtRegistryMutex.Lock()
+	defer stmtRegistryMutex.Unlock()
+
+	// Asssert that all the open statements were closed.
+	for sPtr, query := range openedStmts[c.TestName()] {
+		c.Check(closedStmts[c.TestName()][sPtr], Equals, true,
+			Commentf("%s: failed to close statement: %s", c.TestName(), query))
 	}
 
+	// Reset state for next test.
```

Comment:
Should we still be doing this if these maps are indexed by the tests? If tests are run in parallel for whatever reason I think this could break it.

They could be cleared on suite teardown though.